shelley-ma/impl/src/Cardano/Ledger/Mary/Translation.hs

```diff
@@ -38,11 +60,25 @@ type instance PreviousEra (MaryEra c) = AllegraEra c
 type instance TranslationContext (MaryEra c) = ()
 
 instance Crypto c => TranslateEra (MaryEra c) NewEpochState where
-  translateEra _ = error "TODO Allegra to Mary translation"
+  translateEra ctxt nes =
+    return $
+      NewEpochState
+        { nesEL = nesEL nes,
+          nesBprev = translateEra' ctxt $ nesBprev nes,
+          nesBcur = translateEra' ctxt $ nesBcur nes,
+          nesEs = translateEra' ctxt $ nesEs nes,
+          nesRu = translateEra' ctxt <$> nesRu nes,
+          nesPd = nesPd nes
+        }
 
 instance Crypto c => TranslateEra (MaryEra c) Tx where
-  type TranslationError (MaryEra c) Tx = ()
```

Comment:
You could remove the same line in the Allegra translation too, as that is total as well.